tripy/tripy/frontend/trace/ops/base.py

```diff
@@ -194,6 +204,25 @@ def to_flat_ir(self, inputs: List["FlatIRTensor"], outputs: List["FlatIRTensor"]
         """
         ...
 
+    def convert_to_flat_ir(
+        self, inputs: List["FlatIRTensor"], outputs: List["FlatIRTensor"], flat_ir: Optional["FlatIR"] = None
+    ) -> None:
+        """
+        Convert the trace operation to Flat IR representation.
+
+        This method decides whether to call the wrapped or unwrapped version
+        of to_flat_ir based on the is_flat_ir_conversion_wrapped flag.
+
+        Args:
+            inputs: List of input FlatIRTensor objects
+            outputs: List of output FlatIRTensor objects
+            flat_ir: Optional FlatIR object for the wrapped version
+        """
+        if self.is_to_flat_ir_wrapped:
+            self.to_flat_ir(inputs, outputs, flat_ir)
+        else:
+            self.to_flat_ir(inputs, outputs)
```

Comment:
Can we just do this in a try-catch so we don't need the `is_to_flat_ir_wrapped`?

Reply:
Good idea.